components/vercel/vercel.app.mjs

```diff
@@ -1,11 +1,161 @@
+import { axios } from "@pipedream/platform";
+import constants from "./constants.mjs";
+
 export default {
   type: "app",
   app: "vercel",
-  propDefinitions: {},
+  propDefinitions: {
+    project: {
+      type: "string",
+      label: "Project",
+      description: "Filter deployments from the given projectId",
+      optional: true,
+      async options() {
+        const projects = await this.listProjects();
+        if (!projects || projects.length === 0) {
+          return [];
+        }
+        return projects.map((project) => ({
+          label: project.name,
+          value: project.id,
+        }));
+      },
+    },
+    deployment: {
+      type: "string",
+      label: "Deployment",
+      description: "Select the deployment to cancel",
+      async options({ state }) {
+        const params = state
+          ? {
+            state,
+          }
+          : undefined;
+        const deployments = await this.listDeployments(params);
+        if (!deployments || deployments.length === 0) {
+          return [];
+        }
+        return deployments.map((deployment) => ({
+          label: deployment.name,
+          value: deployment.uid,
+        }));
+      },
+    },
+    team: {
+      type: "string",
+      label: "Team",
+      description: "The Team identifier or slug to perform the request on behalf of",
+      optional: true,
+      async options() {
+        try {
```

Comment:
Is there a reason for this?
```
try {
  ...
} catch (e) {
  throw new Error(e);
}

Reply:
Updated to:
`throw new Error(e.message);`
This is because the `teams` props only works for paid accounts and displays an error otherwise. If I throw `e.message`, it will display "Request failed with status code 403" instead of "Cannot read properties of undefined (reading 'message')"